nautobot/utilities/forms/forms.py

```diff
@@ -233,3 +244,107 @@ def __init__(self, table, *args, **kwargs):
     @property
     def table_name(self):
         return self.table.__class__.__name__
+
+
+class DynamicFilterForm(BootstrapMixin, forms.Form):
+    """
+    Form for dynamically inputting filter values for an object list.
+    """
+
+    lookup_field = forms.ChoiceField(
+        choices=[],
+        required=False,
+        label="Field",
+    )
+    lookup_type = forms.ChoiceField(
+        choices=[],
+        required=False,
+    )
+    lookup_value = forms.CharField(
+        required=False,
+        label="Value",
+    )
+
+    def __init__(self, *args, filterset=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        from nautobot.utilities.forms import add_blank_choice  # Avoid circular import
+
+        # cls.model is set at `dynamic_formset_factory()`
+        self.filterset = filterset or getattr(self, "filterset", None)
+
+        # Raise exception if `cls.filterset` not set and `filterset` not passed
+        if self.filterset is None:
+            raise AttributeError("'DynamicFilterForm' object requires `filterset` attribute")
+
+        model = self.filterset._meta.model
+
+        if self.filterset is not None:
+            filterset = self.filterset()
```

Comment:
Should `cls.filterset` / the `filterset=...` parameter be renamed to `filterset_class` for clarity?